detect/gcp/gce.go

```diff
@@ -0,0 +1,116 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"cloud.google.com/go/compute/metadata"
+
+	"go.opentelemetry.io/otel/api/kv"
+	"go.opentelemetry.io/otel/api/standard"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+// GCE collects resource information of GCE computing instances
+type GCE struct{}
+
+// Detect detects associated resources when running on GCE hosts.
+func (gce *GCE) Detect(ctx context.Context) (*resource.Resource, error) {
+	if !metadata.OnGCE() {
+		return nil, nil
+	}
+
+	labels := []kv.KeyValue{
+		standard.CloudProviderGCP,
+		standard.CloudRegionKey.String(""),
```

Comment:
Should this be added as an empty value? This seems incorrect.

Reply:
I followed this example.  https://github.com/census-ecosystem/opencensus-go-resource/blob/master/gcp/gcp.go#L50

Reply:
That example looks incorrect, it seems like having the actual region would be what users would expect.

Possibly something like:

```
strings.Join(strings.SplitN(metadata.Zone(), "-", 3)[0:2], "-")
```

With some error checking on the return values (instead of just assuming the split returned a full split happened).